ydb/docs/en/core/reference/ydb-cli/parameterized-query-execution.md

```diff
@@ -0,0 +1,604 @@
+
+You can terminate the command by Ctrl+C or wait 200 seconds until the input generation is finished.
+
+#### Limit on the number of records {#example-adaptive-limit}
+
+This example demonstrates the adaptive batching triggered by a number of parameter sets. In the first line of the command below, we generate 200 rows. The command will show parameter batches in each subsequent query call, applying the given limit `--input-batch-max-rows` of 20 (the default limit is 1,000).
+
+In this example, we also demonstrate the option to join parameters from different sources and generate JSON at the output.
+
+```bash
+for i in $(seq 1 200);do echo "Line$i";done | \
+{{ ydb-cli }} -p quickstart sql \
+  -s 'DECLARE $x AS List<Utf8>;
+      DECLARE $p2 AS Int64;
+      SELECT ListLength($x) AS count, $p2 AS p2, $x AS items' \
+  --input-framing newline-delimited \
+  --input-format raw \
+  --input-param-name x \
+  --input-batch adaptive \
+  --input-batch-max-rows 20 \
+  --param '$p2=10' \
+  --format json-unicode
+```
+
+Command output:
+
+```text
+{"count":20,"p2":10,"items":["Line1","Line2","Line3","Line4","Line5","Line6","Line7","Line8","Line9","Line10","Line11","Line12","Line13","Line14","Line15","Line16","Line17","Line18","Line19","Line20"]}
+{"count":20,"p2":10,"items":["Line21","Line22","Line23","Line24","Line25","Line26","Line27","Line28","Line29","Line30","Line31","Line32","Line33","Line34","Line35","Line36","Line37","Line38","Line39","Line40"]}
+...
+{"count":20,"p2":10,"items":["Line161","Line162","Line163","Line164","Line165","Line166","Line167","Line168","Line169","Line170","Line171","Line172","Line173","Line174","Line175","Line176","Line177","Line178","Line179","Line180"]}
+{"count":20,"p2":10,"items":["Line181","Line182","Line183","Line184","Line185","Line186","Line187","Line188","Line189","Line190","Line191","Line192","Line193","Line194","Line195","Line196","Line197","Line198","Line199","Line200"]}
+```
+
+#### Deleting multiple records from a {{ ydb-short-name }} table based on primary keys {#example-adaptive-delete-pk}
+
+{% include [not_allow_for_olap_note](../../_includes/not_allow_for_olap_note.md) %}
+
+This example shows how you can delete an unlimited number of records from {{ ydb-short-name }} tables without risking exceeding the limit on the number of records per transaction.
+
+Let's create a test table:
+
+```bash
+{{ ydb-cli }} -p quickstart sql -s 'create table test_delete_1( id UInt64 not null, primary key (id))'
+```
+
+Add 100,000 records to it:
+
+```bash
+for i in $(seq 1 100000);do echo "$i";done | \
+{{ ydb-cli }} -p quickstart import file csv -p test_delete_1
+```
+
+Delete all records with ID > 10:
+
+```bash
+{{ ydb-cli }} -p quickstart sql \
+  -s 'SELECT t.id FROM test_delete_1 AS t WHERE t.id > 10' \
+  --format json-unicode | \
+{{ ydb-cli }} -p quickstart sql \
+  -s 'DECLARE $lines AS List<Struct<id:UInt64>>;
+      DELETE FROM test_delete_1 WHERE id IN (SELECT tl.id FROM AS_TABLE($lines) AS tl)' \
```

Comment:
This requires extra explanation on why this is so much hassle and not just `DELETE FROM test_delete_1 WHERE id > 10;`